### DISCUSSION TITLE
The future of System Application

### DISCUSSION THREAD

**Discussion by Lazys** [microsoft/ALAppExtensions#21012] [General] [answered]
Hi, just out of curiosity - in the base application we still have some, lets say, generic tools like:
1. CU "Confirm Management",
2. CU "Progress Dialog",
3. CU "Type Helper".

Are you going to move these tools to the System Application layer?

Basically do you have plans that Base Application will contain only logic related to system functionality (processes like Production orders, Sales orders etc.)? Or my understanding is not really correct and the Base Application will be still a mix of different things?

**Comment by mazhelez** [comment, marked as answer]
Hey @Lazys 

All the mentioned functionalities are great candidates for _System Application_'s modules.
_Confirm Management_ is even present there, see [here](https://github.com/microsoft/ALAppExtensions/tree/main/Modules/System/Confirm%20Management).

Without binding to any concrete plans, I can only mention that the overall plan for _System Application_ is to strip down the basic functionalities from _Base Application_ (such as the ones you mentioned) and place them in modules that are well-documented and with clear APIs.

**Reply by Lazys** [reply]
Hi, thanks for the reply.

I think that this is a great direction, so keep moving forward :)

P.S.: the Confirm Management was my mistake, because as you mention it's already extracted to the system application - but I remember the time when it was in the base app :)

Thanks again!
